app/controllers/projects_controller.rb

```diff
@@ -104,48 +103,27 @@ def show
 
   def edit
     project
-    if project.metadata[:status] != Project::APPROVED_STATUS
+    if project.metadata_model.status != Project::APPROVED_STATUS
       flash[:notice] = "Pending projects can not be edited."
       redirect_to project
-    elsif project.metadata[:status] == Project::APPROVED_STATUS && !eligible_editor? #check if the current user is a sponsor of manager
+    elsif project.metadata_model.status == Project::APPROVED_STATUS && !eligible_editor? #check if the current user is a sponsor of manager
```

Comment:
This is not really part of your PR, but since you are touching this line might as well fix the typo
```suggestion
    elsif project.metadata_model.status == Project::APPROVED_STATUS && !eligible_editor? #check if the current user is a sponsor or a manager
```